src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectTracker.cs

```diff
@@ -916,10 +917,35 @@ private void OutputListToOutputWindow(string prefix, IEnumerable<string> values)
                     continue;
                 }
 
-                if (addedReferencePaths.Add(path))
+                if (metadataReferencesToAdd.TryGetValue(path, out var existingProperties))
                 {
-                    projectContext.AddMetadataReference(path, reference.Properties);
+                    // Merge existing aliases the properties that are already there
+                    var allAliases = existingProperties.Aliases.AddRange(reference.Properties.Aliases);
+
+                    // If one is empty and the other isn't, then we need to toss the global alias in too. The other cases
```

Comment:
> If one is empty and the other isn't, then we need to toss the global alias in too.

Why? It seems strange to me that we have to manage the global alias. Shouldn't that always be available, regardless of what we do?